MQ2AutoSize.cpp

```diff
@@ -359,269 +536,1087 @@ PLUGIN_API void OnPulse()
 
-	// TODO:  Previous implementation of toggle by range was broken.  Need to add handling for toggle when range is set
 	if (!*szCurArg)
 	{
-		SetEnabled(!AS_Config.OptByZone);
+		// fake Zonewide with large value which is effectively Far Cliping Plane at 100%
+		if (AS_Config.ResizeRange >= MIN_SIZE && AS_Config.ResizeRange < FAR_CLIP_PLANE)
+		{
+			// this means we are currently using Range distance normally
+			// now we want look like we are toggling to Zonewide
+			// we will do this by increasing the ResizeRange to FAR_CLIP_PLANE
+			emulate("zonewide");
+		}
+		else if (AS_Config.ResizeRange == FAR_CLIP_PLANE)
+		{
+			// this means we are pretending to be Zonewide and need to revert to Range based
+			// now we want look like we are toggling to Zonewide
+			// we will do this by reseting the ResizeRange to what it was prior to going zonewide
+			emulate("range");
+		}
+		return;
 	}
-	else
+	else if (ci_equals(szCurArg, "dist"))
 	{
-		// TODO:  else after returns here, determine what's actually needed and clean up
-		if (!_strnicmp(szCurArg, "dist", 5))
+		if (ci_equals(szNumber, "on"))
 		{
-			AS_Config.OptByRange = !AS_Config.OptByRange;
-			if (AS_Config.OptByRange)
+			if (AS_Config.ResizeRange == FAR_CLIP_PLANE)
 			{
-				if (AS_Config.OptByZone)
-				{
-					AS_Config.OptByZone = false;
-					WriteChatf("\ay%s\aw:: AutoSize (\ayAllZone\ax) now \ardisabled\ax!", MODULE_NAME);
-				}
+				emulate("range");
 			}
-			else
+		}
+		else if (ci_equals(szNumber, "off"))
+		{
+			if (AS_Config.ResizeRange != FAR_CLIP_PLANE)
 			{
-				SpawnListResize(true);
+				emulate("zonewide");
 			}
-			WriteChatf("\ay%s\aw:: AutoSize (\ayRange\ax) now %s\ax!", MODULE_NAME, AS_Config.OptByRange ? "\agenabled" : "\ardisabled");
-			if (AS_Config.OptAutoSave) SaveINI();
-			return;
 		}
-		else if (!_strnicmp(szCurArg, "save", 5))
+		else if (!ci_equals(szNumber, "on") && !ci_equals(szNumber, "off"))
 		{
-			SaveINI();
-			WriteChatf("\ay%s\aw:: Configuration file saved.", MODULE_NAME);
-			return;
+			if (AS_Config.ResizeRange == FAR_CLIP_PLANE)
+			{
+				emulate("range");
+			}
+			else if (AS_Config.ResizeRange != FAR_CLIP_PLANE)
+			{
+				emulate("zonewide");
+			}
 		}
-		else if (!_strnicmp(szCurArg, "load", 5))
+		return;
+	}
+	else if (ci_equals(szCurArg, "save"))
+	{
+		SaveINI();
+		return;
+	}
+	else if (ci_equals(szCurArg, "load"))
+	{
+		LoadINI();
+		return;
+	}
+	else if (ci_equals(szCurArg, "autosave"))
+	{
+		if (ci_equals(szNumber, "on"))
 		{
-			LoadINI();
-			WriteChatf("\ay%s\aw:: Configuration file loaded.", MODULE_NAME);
-			return;
+			if (!AS_Config.OptAutoSave)
+			{
+				ToggleOption("Autosave", &AS_Config.OptAutoSave);
+			}
+		}
+		else if (ci_equals(szNumber, "off"))
+		{
+			if (AS_Config.OptAutoSave)
+			{
+				ToggleOption("Autosave", &AS_Config.OptAutoSave);
+			}
 		}
-		else if (!_strnicmp(szCurArg, "autosave", 9))
+		else if (!ci_equals(szNumber, "on") && !ci_equals(szNumber, "off"))
```

Comment:
same as above. You can't get here if szNumber was on or off, so condition isn't necessary.